kms/templates/cloudtrail.json.tpl

```diff
@@ -88,6 +88,23 @@
           "kms:EncryptionContext:aws:cloudtrail:arn": "arn:aws:cloudtrail:*:${account_id}:trail/*"
         }
       }
+    },
+    {
+      "Sid": "Allow S3 logs to send events to encrypted SNS topic",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "s3.amazonaws.com"
+      },
+      "Action": [
+        "kms:GenerateDataKey",
+        "kms:Decrypt"
```

Comment:
Might be a silly question, but why does this policy need Decrypt permissions rather than Encrypt? Isn't Cloudtrail doing the encryption step?

Reply:
It's in the link I put in the other comments. S3 is the producer, sending messages to the SNS queue and it says 

> the producer's next call to SendMessage or SendMessageBatch also triggers calls to kms:GenerateDataKey and kms:Decrypt. The call to kms:Decrypt is to verify the integrity of the new data key before using it.

Reply:
I missed that, thanks 👍